backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/Doc_AllocationHdr.java

```diff
@@ -944,15 +949,24 @@ private AmountSourceAndAcct createPurchaseSalesInvoiceFacts(final Fact fact, fin
 				.orgId(counterLine.getInvoiceOrgId())
 				.bPartnerAndLocationId(counterLine.getInvoiceBPartnerId(), counterLine.getInvoiceBPartnerLocationId())
 				.alsoAddZeroLine();
+
 		if (counterLine.isSOTrxInvoice())
 		{
 			factLineBuilder.setAccount(getCustomerAccount(BPartnerCustomerAccountType.C_Receivable, as));
 			factLineBuilder.setAmtSource(null, compensationAmtSource.negate());
 		}
 		else
 		{
-			factLineBuilder.setAccount(getVendorAccount(BPartnerVendorAccountType.V_Liability, as));
-			factLineBuilder.setAmtSource(compensationAmtSource, null);
+			if(counterLine.isService() && line.isCreditMemoInvoice())
+			{
+				factLineBuilder.setAccount(getVendorAccount(BPartnerVendorAccountType.V_Liability, as));
```

Comment:
the setAccount(V_Liability) is the same for if and else. move it up, above the if